dev/dev-env-setup.sh

```diff
@@ -145,189 +148,215 @@ version_gt() {
+}
 
 # Check if the virtualenv already exists at the specified path
-if [[ -d "$directory"  ]]; then
-  if [ -z "$GITHUB_ACTIONS" ]; then
-    read -p "A virtual environment is already located at $directory. Do you wish to replace it? $(tput bold; tput setaf 2)(y/n) $(tput sgr0)" -n 1 -r
-    echo
-  fi
-  if [[ $REPLY =~ ^[Yy]$ || -n "$GITHUB_ACTIONS" ]]; then
-    echo "Replacing Virtual environment in '$directory'. Installing new instance."
-    pyenv exec virtualenv --clear "$directory"
+create_virtualenv() {
+  if [[ -d "$directory" ]]; then
+    if [ -z "$GITHUB_ACTIONS" ]; then
+      read -p "A virtual environment is already located at $directory. Do you wish to replace it? $(
+        tput bold
+        tput setaf 2
+      )(y/n) $(tput sgr0)" -n 1 -r
+      echo
+    fi
+    if [[ $REPLY =~ ^[Yy]$ || -n "$GITHUB_ACTIONS" ]]; then
+      echo "Replacing Virtual environment in '$directory'. Installing new instance."
+      pyenv exec virtualenv --clear "$directory"
+    fi
+  else
+    # Create a virtual environment with the specified Python version
+    pyenv exec virtualenv --python "$PY_INSTALL_VERSION" "$directory"
   fi
-else
-  # Create a virtual environment with the specified Python version
-  pyenv exec virtualenv --python "$PY_INSTALL_VERSION" "$directory"
-fi
 
-# Activate the virtual environment
-# shellcheck disable=SC1090
-source "$VENV_DIR"
+  # Activate the virtual environment
+  # shellcheck disable=SC1090
+  source "$VENV_DIR"
 
-echo "$(tput setaf 2)Current Python version: $(tput bold)$(python --version)$(tput sgr0)"
-echo "$(tput setaf 3)Activated environment is located: $(tput bold) $directory/bin/activate$(tput sgr0)"
-
-echo "Installing pip dependencies for development environment."
+  echo "$(tput setaf 2)Current Python version: $(tput bold)$(python --version)$(tput sgr0)"
+  echo "$(tput setaf 3)Activated environment is located: $(tput bold) $directory/bin/activate$(tput sgr0)"
+}
 
 # Install current checked out version of MLflow (local)
-pip install $(quiet_command) -e .[extras]
-pip install pre-commit
-
-if [[ -n "$full" ]]; then
-  # Install dev requirements and test plugin
-  pip install $(quiet_command) -r "$rd/dev-requirements.txt"
-  # Install test plugin
-  pip install $(quiet_command) -e "$MLFLOW_HOME/tests/resources//mlflow-test-plugin"
-  echo "Finished installing pip dependencies."
-else
-  files=("$rd/test-requirements.txt" "$rd/lint-requirements.txt" "$rd/doc-requirements.txt")
-  for r in "${files[@]}";
-  do
-    pip install $(quiet_command) -r "$r"
-  done
-fi
-
-echo "$(tput setaf 2; tput smul)Python packages that have been installed:$(tput rmul)"
-echo "$(pip freeze)$(tput sgr0)"
+install_mlflow() {
+  echo "Installing pip dependencies for development environment."
+
+  pip install $(quiet_command) -e .[extras]
+  pip install pre-commit
+
+  if [[ -n "$full" ]]; then
+    # Install dev requirements and test plugin
+    pip install $(quiet_command) -r "$rd/dev-requirements.txt"
```

Comment:
I'm not sure how quiet is useful. It's useful when we need to run this script on a machine where the amount of logs is limited, otherwise we should not silence logs. Can w e remove it? @BenWilson2 WDYT?

Reply:
It's fine to remove it. When some packages were still having to build and compile from source when we first built this script it was useful, but now that TF, Torch, LightGBM, and XGBoost are all supported on OSX METAL, the pip install commands are nowhere near as verbose as they used to be (IIRC, the build-from-source compile for TF used to generate >20k lines of stdout messaging)

Reply:
sounds good